gamepad/gamepad_darwin.go

```diff
@@ -0,0 +1,155 @@
+package gamepad
+
+/*
+#cgo CFLAGS: -Werror -xobjective-c -fmodules -fobjc-arc
+
+#import <Foundation/Foundation.h>
+#import <GameController/GameController.h>
+
+static CFTypeRef getGamepads() {
+	if (@available(iOS 15, macOS 12, *)) {
+		NSArray<GCController *> * Controllers = [GCController controllers];
+		return (CFTypeRef)CFBridgingRetain(Controllers);
+	}
+	return 0;
+}
+
+static CFTypeRef getState(CFTypeRef gamepads, int64_t player) {
+	if (@available(iOS 15, macOS 12, *)) {
+		NSArray<GCController *> * Controllers = (__bridge NSArray<GCController *> *)gamepads;
+		if ([Controllers count] <= player) {
+			return 0;
+		}
+
+		GCExtendedGamepad * Gamepad = [[Controllers objectAtIndex:player] extendedGamepad];
+		if (Gamepad == nil) {
+			return 0;
+		}
+
+		GCPhysicalInputProfile* Inputs = (GCPhysicalInputProfile*)Gamepad;
+		return (CFTypeRef)CFBridgingRetain(Inputs);
+	}
+	return 0;
+}
+
+static double getLastEventFrom(CFTypeRef inputs) {
+	if (@available(iOS 15, macOS 12, *)) {
+		return (double)(((__bridge GCPhysicalInputProfile*)(inputs)).lastEventTimestamp);
+	}
+	return 0;
+}
+
+static NSString * getKeyName(GCPhysicalInputProfile * Inputs, void * button) {
+	if (@available(iOS 15, macOS 12, *)) {
```

Comment:
For some reason that `@available(iOS 15, macOS 12, *)` not work. I need to change `gogio` to force minSDK to `15` to compile, maybe expose `-minsdk` on gogio fixes the issue and remove that `if`.

Reply:
Fixed by using `API_AVAILABLE(ios(15), macosx(12))`.